### DISCUSSION TITLE
Questions about tracking stock parameters

### DISCUSSION THREAD

**Discussion by icvdok** [inventree/InvenTree#4827] [Q&A]
I've just a question about my case. I'm just started to play a bit with inventree. I'll try to explain, with an actual example. 

Let suppose that I've five RPI4 (same model, rev. etc.) to manage, purchased in two orders , I've created the part  (RPI4) and two PO that create the stock of five items. I would like to keep track the MAC address of each RPI4. 

I've try with parameters but are linked to a part, not usable.
I've try to track the part (making it trackable) adding a specific unique SN but , as far as I know, there is no way to add parameters for each stock item.

Not sure if my case is out of inventree scope or my case have a solution.

Thanks in advance for the support

**Comment by SchrodingersGat** [comment]
You could use the MAC address as the S/N (as you say), or add it into the notes field for the stock item.

However currently key:value parameters for an individual stock item are not available! May be something we can work towards though

**Comment by miggland** [comment]
Stock items have metadata, that would be another option. You can't rely on the existing GUI for this though, you would have to use the API/python binding/create a plugin..

**Reply by SchrodingersGat** [reply]
[Metadata](https://docs.inventree.org/en/latest/extend/plugins/metadata/) is another option but that is really just for plugins.

**Comment by icvdok** [comment]
Thanks you all for the reply and the support, I'll use the note field for the unique information I need to add. I don't have enough developement skill unfortunately.

Maybe in the future it would be a good idea to extend the parameter usage also in stock area.

**Comment by SchrodingersGat** [comment]
> Maybe in the future it would be a good idea to extend the parameter usage also in stock area.

There's currently some work in progress to significantly update the parameters feature - https://github.com/inventree/InvenTree/pull/4823

So maybe after that we can extend to stock items
